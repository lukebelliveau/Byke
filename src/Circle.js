import React from 'react';
import { View } from 'react-native';

export default ({ color }) => <View style={style(color)} />;

const style = color => ({
  width: 20,
  height: 20,
  borderRadius: 20 / 2,
  backgroundColor: color,
});
