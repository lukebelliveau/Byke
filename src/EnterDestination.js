// @flow
import React, { Component } from 'react';
import { View, TextInput, StyleSheet, Platform } from 'react-native';

const initialState = {
  enteredText: '',
};

class EnterDestination extends Component {
  state = initialState;
  constructor(props) {
    super(props);
    this.props = props;
  }

  textChanged = event => {
    this.setState({ enteredText: event });
  };

  render() {
    return (
      <View style={[this.props.style, styles.container]}>
        <TextInput
          style={styles.input}
          underlineColorAndroid="transparent"
          placeholder="Where are you going?"
          placeholderTextColor="lightgray"
          onChangeText={this.textChanged}
          onSubmitEditing={() => this.props.onSubmit(this.state.enteredText)}
          returnKeyType="go"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'dodgerblue',
    flexDirection: 'column-reverse',
    alignItems: 'center',
    shadowColor: 'black',
    shadowRadius: 10,
    shadowOpacity: 100,
    elevation: 10,
  },
  input: {
    flex: Platform.OS === 'ios' ? 1 : 0,
    maxHeight: Platform.OS === 'ios' ? 25 : 50,
    fontSize: 15,
    width: '75%',
    backgroundColor: 'deepskyblue',
    color: 'white',
    borderRadius: 10,
    margin: 20,
    padding: 5,
    paddingLeft: 10,
  },
});

export default EnterDestination;
