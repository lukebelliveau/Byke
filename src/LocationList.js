import React from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  TouchableOpacity,
  Text,
} from 'react-native';

const LocationList = ({ results, onSelect }) => {
  return (
    <View
      style={styles.container}
      pointerEvents={results.length > 0 ? 'auto' : 'none'}
    >
      <ScrollView showsVerticalScrollIndicator={false}>
        {results.map(result =>
          <LocationCard result={result} key={result.id} onSelect={onSelect} />
        )}
      </ScrollView>
    </View>
  );
};

const LocationCard = ({ result, onSelect }) => {
  console.log(result);
  return (
    <View style={styles.card}>
      <View style={styles.infoContainer}>
        <Info name={result.name} address={result.vicinity} />
      </View>
      <View style={styles.buttonContainer}>
        <Button
          onSelect={onSelect}
          coordinates={{
            latitude: result.geometry.location.lat,
            longitude: result.geometry.location.lng,
          }}
        />
      </View>
    </View>
  );
};

const Button = ({ coordinates, onSelect }) => {
  console.log('BUTTON');
  console.log(coordinates);
  return (
    <TouchableOpacity
      style={styles.button}
      onPress={() => onSelect(coordinates)}
    >
      <Text style={styles.buttonText}>
        GO
      </Text>
    </TouchableOpacity>
  );
};

const Info = ({ name, address }) =>
  <View style={{ flex: 1 }}>
    <Text style={styles.title}>{name}</Text>
    <Text style={styles.address}>{address}</Text>
  </View>;

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    display: 'flex',
    top: 0,
    width: '100%',
    height: '100%',
    marginRight: '10%',
    paddingLeft: '5%',
    paddingRight: '5%',
  },
  card: {
    display: 'flex',
    flexDirection: 'row',
    width: '100%',
    height: 150,
    backgroundColor: 'white',
    margin: 5,
    elevation: 5,
  },
  infoContainer: {
    flex: 4,
    paddingLeft: 10,
    paddingTop: 5,
  },
  title: {
    fontSize: 25,
  },
  address: {
    position: 'absolute',
    bottom: 5,
    fontSize: 15,
    width: '75%',
    color: 'gray',
  },
  buttonContainer: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 15,
  },
  button: {
    flex: 1,
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'ghostwhite',
  },
  buttonText: {
    fontSize: 40,
    color: 'deepskyblue',
  },
});

export default LocationList;
