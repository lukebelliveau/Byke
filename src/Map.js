import React from 'react';
import MapView from 'react-native-maps';

import StationMarker from './StationMarker';

const Map = ({ region, stations, style }) => (
    <MapView
        style={style}
        region={region}
        initialRegion={region}>
        <MapView.Marker coordinate={region} />
        {
            stations.map((station, index) => (
                <StationMarker coordinate={station} stationName={station.stationName} availableBikes={station.availableBikes} key={index} />
            ))
        }
    </MapView>
);

export default Map;