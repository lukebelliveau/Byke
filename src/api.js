// @flow
export const url = 'http://byke-graphql.herokuapp.com/graphql?query=';

const googleMapsAPIKey = 'AIzaSyAQhawlRtPkYdpC4VHgaB1n3yLOVOEZbYE';

const getAllStations = (stationParams: string) =>
  fetch(`${url}{allStations${stationParams}}`)
    .then(response => {
      return response.json().then(res => {
        return res.data.allStations;
      });
    })
    .catch(e => console.log(e));

const searchPlaces = (searchQuery: string, latitude, longitude) =>
  fetch(
    `https://maps.googleapis.com/maps/api/place/nearbysearch/json?key=${googleMapsAPIKey}&location=${latitude},${longitude}&radius=10000&keyword=${searchQuery}`
  )
    .then(response => {
      response.json().then(json => json.results);
    })
    .catch(e => console.log(e));

export default {
  getAllStations,
  searchPlaces,
};
