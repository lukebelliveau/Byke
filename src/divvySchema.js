export default`
type Query {
  allStations: [Station]
}

type Station {
  id: ID,
  stationName: String,
  availableDocks: Int,
  totalDocks: Int,
  availableBikes: Int,
  latitude: Float,
  longitude: Float,
}`