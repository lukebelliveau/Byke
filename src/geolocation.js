const defaultPositionOptions: PositionOptions = {
  enableHighAccuracy: true,
  timeout: Math.POSITIVE_INFINITY,
  maximumAge: 0,
};

const defaultError = error => {
  console.log(error);
};

const getLocation = (
  success,
  error = defaultError,
  options = defaultPositionOptions
) => navigator.geolocation.getCurrentPosition(success, error, options);

export default getLocation;
