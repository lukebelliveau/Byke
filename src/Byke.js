// @flow
import React from 'react';
import { StyleSheet, View } from 'react-native';

import api from './api';
import Map from './Map';
import EnterDestination from './EnterDestination';
import getLocation from './geolocation';

import LocationList from './LocationList';
import computeRegion from './utils';

import testLocations from '../test/testLocations';

const initialState = {
  region: null,
  results: [],
  trip: null,
};

class Byke extends React.Component {
  state = initialState;
  constructor() {
    super();

    this.watchLocation();

    this.searchDestination = this.searchDestination.bind(this);
  }

  watchLocation() {
    navigator.geolocation.watchPosition(
      location => {
        this.setState(prevState => ({
          region: {
            latitude: location.coords.latitude,
            longitude: location.coords.longitude,
            latitudeDelta: 0.01,
            longitudeDelta: 0.01,
          },
        }));
      },
      null,
      {}
    );
  }

  componentDidMount() {
    this.centerRegionOnUser();
    // setInterval(() => this.centerRegionOnUser(), 1000);
  }

  destinationSelected = destination => {
    getLocation(location => {
      this.setState({
        region: computeRegion([destination, location.coords]),
        results: [],
        trip: {
          currentLocation: location.coords,
          destination,
        },
      });
    });
  };

  centerRegionOnUser() {
    getLocation(location => {
      this.setState(prevState => ({
        region: {
          latitude: location.coords.latitude,
          longitude: location.coords.longitude,
          latitudeDelta: 0.01,
          longitudeDelta: 0.01,
        },
      }));
    });
  }

  searchDestination = (searchQuery: string) => {
    api
      .searchPlaces(
        searchQuery,
        this.state.region.latitude,
        this.state.region.longitude
      )
      .then(response =>
        response.json().then(json => {
          this.setState({
            results: json.results,
          });
        })
      );
  };

  render() {
    return this.state.region == null
      ? <View />
      : <View style={styles.container}>
          <EnterDestination
            onSubmit={this.searchDestination}
            style={styles.destination}
          />
          <View style={styles.map}>
            <Map region={this.state.region} trip={this.state.trip} />
            <LocationList
              results={this.state.results}
              onSelect={this.destinationSelected}
            />
          </View>
        </View>;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  destination: {
    flex: 1,
  },
  map: {
    flex: 6,
  },
});

export default Byke;
