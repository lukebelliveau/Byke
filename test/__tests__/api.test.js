import React from 'react';

import api from '../../src/api';
import { url as serverURL } from '../../src/api';

import mockGraphQLServer from '../fetchMockGraphQLServer';

describe('graphQL client', () => {
    it('should return the correct params', done => {
        const mocks = {
            Station: () => ({
                latitude: () => 99,
                longitude: () => 99,
            })
        }

        mockGraphQLServer(serverURL, mocks);

        api.getAllStations(`{latitude, longitude}`)
            .then(response => {
                expect(response).toMatchSnapshot();
                done();
            });
    });
})

