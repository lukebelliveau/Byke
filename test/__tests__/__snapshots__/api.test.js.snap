// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphQL client should return the correct params 1`] = `
Array [
  Object {
    "latitude": 99,
    "longitude": 99,
  },
  Object {
    "latitude": 99,
    "longitude": 99,
  },
]
`;
